src/python/pants/task/console_task.py

```diff
@@ -29,7 +31,8 @@ def register_options(cls, register):
 
   def __init__(self, *args, **kwargs):
     super(ConsoleTask, self).__init__(*args, **kwargs)
-    self._console_separator = self.get_options().sep.decode('string-escape')
+    encoding = 'string-escape' if PY2 else 'unicode-escape'
```

Comment:
Minor nit, invert the order so that the Python 3 case comes first.

Reply:
Why?

Reply:
We want to be writing code that's Python 3 first, where Py2 is the edge case we have to make modifications for, rather than the reverse. 

While the functionality is the same either way, inverting the order reflects the goal of Py 3 first.

Reply:
I sorta fixed this issue with https://github.com/pantsbuild/pants/pull/6304/files#diff-5b46183b49344152459937a17ec77085R32, but I think my solution won't work on Py3 so a mix of the two works best.

Specifically, I think you're right to first call `.encode('utf-8')` and then to decode. In Py3, (and with this patch?), `sep` is unicode so Py3 will complain about decoding unicode. 

I would maybe keep the change I did to always use `unicode_escape`. It works in Py2, and reduces complexity to be using the same semantics in both versions.

You'll need to keep the change I made to then yet again encoding in `execute`, because I changed `_outstream` to be bytes file.

--

Alternatively, if we only use `_console_separator` in `execute()`, which looks to be true—i.e. we don't care about `_console_separator` being unicode—we can simplify this all a lot and just do `self.get_options().sep.encode('unicode_escape')` to get rid of all this boilerplate.